crates/cubecl-linalg/src/matmul/components/global/tensor_view.rs

```diff
@@ -25,8 +24,8 @@ pub struct TensorReader<GA: MatmulArgs<E>, E: Numeric> {
 /// A view of a tensor that starts reading data from a specified offset.
 /// Ensures safe access by preventing out-of-bounds errors.
 /// Includes pre-fetched shapes and strides for optimized performance.
-pub struct TensorWriter<GA: MatmulArgs<E>, E: Numeric> {
-    pub tensor: TensorOutput<E, GA>,
+pub struct TensorWriter<E: Numeric> {
+    pub tensor: VirtualTensor<E, ReadWrite>,
```

Comment:
To save a little bit of duplication, we could use the newly introduced IO concept and define a struct

```rust
pub struct TensorIO<E, IO> { }
```

with possibly type aliases to avoid updating the rest of the code base.

```rust
pub type TensorReader<E> = TensorIO<E, Read>;
pub type TensorWriter<E> = TensorIO<E, ReadWrite>;
```